File: 9-PalindromeNumber/9-PalindromeNumber.js
// Last updated: 24/04/2026, 09:57:36
var isPalindrome = function(x) {
    if (x < 0) return false;   // negative numbers are not palindrome

    let original = x;
    let reversed = 0;

    while (x > 0) {
        let digit = x % 10;
        reversed = reversed * 10 + digit;
        x = Math.floor(x / 10);
    }

    return original === reversed;
};

File: 9-PalindromeNumber/9-PalindromeNumber.cpp
// Last updated: 24/04/2026, 09:57:28
class Solution {
public:
    bool isPalindrome(int x) {
        if(x < 0) return false;

        int original = x;
        long long sum = 0;  // 👈 changed to long long

        while(x > 0) {
            int rem = x % 10;
            sum = sum * 10 + rem;
            x = x / 10;
        }

        return sum == original;
    }
};